lib/github-client.js

```diff
@@ -1,45 +1,71 @@
 /* jshint node: true */
 'use strict';
 
-var GitHubApi = require('github');
+var http = require('http');
+var https = require('https');
+var packageInfo = require('../package.json');
 
 module.exports = GitHubClient;
 
-function GitHubClient(config, githubApi) {
+function GitHubClient(config) {
   this.user = config.githubUser;
-  this.api = githubApi || createGitHubApi(config);
+  this.timeout = config.githubTimeout;
+  this.protocol = 'https:';
+  this.host = 'api.github.com';
 }
 
-function createGitHubApi(config) {
-  var api = new GitHubApi({
-    version: '3.0.0',
-    protocol: 'https',
-    timeout: config.githubTimeout
-  });
-
-  api.authenticate({
-    type: 'oauth',
-    token: process.env.HUBOT_GITHUB_TOKEN
-  });
-  return api;
-}
+GitHubClient.prototype.httpOptions = function(repository, paramsStr) {
+  return {
+    protocol: this.protocol,
+    host: this.host,
+    port: this.port,
+    path: '/repos/' + this.user + '/' + repository + '/issues',
+    method: 'POST',
+    headers: {
+      'Accept': 'application/vnd.github.v3+json',
+      'Authorization': 'token ' + process.env.HUBOT_GITHUB_TOKEN,
+      'Content-Type': 'application/json',
+      'Content-Length': Buffer.byteLength(paramsStr, 'utf8'),
+      'User-Agent': packageInfo.name + '/' + packageInfo.version
+    }
+  };
+};
```

Comment:
It replaces the [github npm](https://npmjs.org/packages/github) that I was using. I think the problem I hit was one of user permissions, but I didn't realize it until after I wrote this. On top of that, it's nice to cut dependencies, and I hope to get some pedagogical mileage out of it in https://github.com/18F/unit-testing-node/.